### ISSUE TITLE
🐛 Bug: Renovate merges PRs with failing migration tests when release flow unlocks branch protections (I think)

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [JoshuaKGoldberg/create-typescript-app#1742]
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/create-typescript-app/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

Renovate PRs with failing CI statuses should not be merged into `main`.

### Actual

Look at all these PRs whose automerges merged with ❌ CI:

* #1728
* #1731
* #1738

In addition to being annoying for me to have to keep reverting the PRs, I think it's also an indication that the current release flow is flawed. Right now the release flow is:

1. Save current `main` branch protections in memory
2. Delete `main` branch protections on GitHub
3. Run the release
4. Restore `main` branch protections on GitHub from memory

Between 2 and 4 is a general opening for squash merging to happen in this repo - and any downstream repo with a similar setup. Not good.

Or at least I think. I'm not 100% sure. But this is my current best theory and strikes me as reasonable.

### Additional Info

I suspect the right solution will be, rather than _deleting_ branch protections, to instead _change_ them surgically to allow the running release user. That way they still block apps like Dependabot and Renovate.

In other words:

1. Save current `main` branch protections in memory
2. Change `main` branch protections on GitHub to exempt the release bot/user
3. Run the release
4. Restore `main` branch protections on GitHub from memory

💖

**Comment by JoshuaKGoldberg** [comment]
#1747 didn't help, they're still being pinned.

**Comment by JoshuaKGoldberg** [comment]
Well, as of #1788, this repo and template no longer use branch protections. So this issue is aged away. It's roughly replaced by another irksome issue, #1838.
